### ISSUE TITLE
Issues to introduce intermediate configuration

### ISSUE THREAD

**Issue by Simhano** [deepmodeling/jax-fem#48]
Hi, JAX-FEM community,
Now, I am trying to introduce an intermediate configuration to optimize the initial coordinate of geometry. (To find the gradient in terms of the initial coordinate.)

For example, the result is not zero. and different from changing params (So, the problem is depends on params at least.):
```python
params_1 = original_cood * 0
params_2 = original_cood*0.02
sol_list_1 = fwd_pred(params_1)
sol_list_2 = fwd_pred(params_2)
print("Solution difference (params_1 vs params_2):", np.linalg.norm(sol_list_1[0] - sol_list_2[0]))
```

But the differentiation is giving me zeros. (d_coord)
```python
def test_fn(sol_list):
    #print('test fun')
    # jax.debug.print("cost func: {}", np.sum((sol_list[0] - u_sol_2)**2))
    return np.sum((sol_list[0] - u_sol_2)**2)*10000 #np.sum((sol_list[0] - u_sol_2)**2)

def composed_fn(params):
    # print()
    return test_fn(fwd_pred(params))

d_coord= jax.grad(composed_fn)(params)
```
It seems like the 'jax. grad(composed_fn)(params)' calculation finished after one iteration. (The test function value is not zero.)

Does anyone have any idea what I need to fix more or something I missed? or some places I need to check?

I am not very familiar with coding with JAX so my code and question can be not clear or well organized. I am sorry about that.
Thank you!

**Comment by tianjuxue** [comment]
This feature of taking derivatives w.r.t mesh coordinates is now available. Please refer to [this example](https://github.com/deepmodeling/jax-fem/tree/a2202e3b698f8c087d3e9f37d7ea0d0d38debd99/applications/differentiable_mesh).
